pkg/apis/app/v1beta1/types.go

```diff
@@ -220,6 +221,7 @@ const (
 	FlinkApplicationSubmittingJob   FlinkApplicationPhase = "SubmittingJob"
 	FlinkApplicationRunning         FlinkApplicationPhase = "Running"
 	FlinkApplicationSavepointing    FlinkApplicationPhase = "Savepointing"
+	FlinkApplicationCancelling      FlinkApplicationPhase = "Cancelling"
```

Comment:
Can you also update state_machine.md with the details of this new state?

Reply:
done!